### PR TITLE
Added Debian package build dir

### PR DESCRIPTION
**Pull request by balena**
I left a debian folder containing the needed files to generate a librocksdb4 and a librocksdb4-dev packages.

### CODE REVIEW

**Comment by balena** [comment]
Actually I have made other polishings on the debian package, specially 
regarding the PORTABLE=1 flag (I've got a SIGILL after copying a 
generated .deb file from a machine to another).

Contributor License Agreement signed.

-- 
Guilherme B. Versiani

File: debian/copyright [hunk 1]
```diff
@@ -0,0 +1,61 @@
+Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: rocksdb
+Upstream-Contact: G. B. Versiani <guibv@yahoo.com>
+Source: http://rocksdb.org
+
+Files: debian/*
+Copyright:
+  2015 G. B. Versiani <guibv@yahoo.com>
+License: GPL-3
```
Comment (adamretter) [inline debian/copyright]: Does this really need to be GPL-3? I think it would be better not to introduce another licence (beyond BSD)

File: Makefile [hunk 2]
```diff
@@ -999,30 +999,30 @@ ldb: tools/ldb.o $(LIBOBJECTS)
 INSTALL_PATH ?= /usr/local
 
 uninstall:
-	rm -rf $(INSTALL_PATH)/include/rocksdb \
-	  $(INSTALL_PATH)/lib/$(LIBRARY) \
-	  $(INSTALL_PATH)/lib/$(SHARED4) \
-	  $(INSTALL_PATH)/lib/$(SHARED3) \
-	  $(INSTALL_PATH)/lib/$(SHARED2) \
-	  $(INSTALL_PATH)/lib/$(SHARED1)
+	rm -rf $(DESTDIR)$(INSTALL_PATH)/include/rocksdb \
```
Comment (adamretter) [inline Makefile]: Why the change from `INSTALL_PATH` to `DESTDIR`?

File: debian/control [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Maintainer: G. B. Versiani <guibv@yahoo.com>
+Priority: optional
+Standards-Version: 3.9.5
+Build-Depends: debhelper (>> 9),
+ devscripts,
+ hardening-includes
+Section: libs
+Homepage: http://rocksdb.org
+
+Package: librocksdb4.4-dev
```
Comment (adamretter) [inline debian/control]: Do we have to hardcode this as 4.4? Is it possible to have a script that updates the version number to the current version?

File: debian/copyright (hunk 1 above)
Comment (balena) [inline debian/copyright]: My fault. Single line change. Should be BSD indeed.

File: Makefile (hunk 2 above)
Comment (balena) [inline Makefile]: A bit of personal taste and GNU Coding Standards: https://www.gnu.org/prep/standards/html_node/DESTDIR.html. It also eases interfacing with the Debian build. Not doing so forces builders to customize their builds (which are usually adjusted after the above standards).

File: debian/control (hunk 3 above)
Comment (balena) [inline debian/control]: I've noted some compatibility problems with the library along the time, so upgrades may affect its dependents (this is one: https://github.com/stephan-hof/pyrocksdb). So, I wanted to stay on the safe side and keep two version numbers. But you, as maintainers, will always have more control over the changes, and may opt to use a single version number when you know recent changes don't break the API.
